src/lib/phdi-building-blocks/phdi_building_blocks/schemas.py

```diff
@@ -0,0 +1,251 @@
+
+    :param value: A list containing the values stored in a top level key in a FHIR
+    resource.
+    :param selection_criteria: A string indicating which element of list to select
+    when one is encountered during parsing.
+    """
+
+    if selection_criteria == "first":
+        value = value[0]
+    elif selection_criteria == "last":
+        value = value[-1]
+    elif selection_criteria == "random":
+        value = random.choice(value)
+    elif selection_criteria == "all":
+        value = value
+
+    # Temporary hack to ensure no structured data is written using pyarrow.
+    # Currently Pyarrow does no support mixing non structure and structured data.
+    # https://github.com/awslabs/aws-data-wrangler/issues/463
+    # Will need to consider other methods of writing to parquet if this is and essential
+    # feature.
+    if type(value) == dict:
+        value = json.dumps(value)
+    elif type(value) == list:
+        value = ",".join(value)
+    return value
+
+
+def apply_schema_to_resource(resource: dict, schema: dict) -> dict:
+    """
+    Given a resource and a schema return a dictionary with values of the data
+    specified by the schema and associated keys defined by the variable name provided
+    by the schema.
+
+    :param resource: A FHIR resource on which to apply a schema.
+    :param schema: A schema specifying the desired values by FHIR resource type.
+    :return data: A dictionary containing the data extracted from the patient a long
+    with specified variable names.
+    """
+
+    data = {}
+    resource_schema = schema.get(resource.get("resourceType", ""))
+    if resource_schema is None:
+        return data
+    for field in resource_schema.keys():
+
+        path = resource_schema[field]["fhir_path"]
+        value = evaluate(resource, path)
+
+        if len(value) == 0:
+            data[resource_schema[field]["new_name"]] = ""
+        else:
+            selection_criteria = resource_schema[field]["selection_criteria"]
+            value = apply_selection_criteria(value, selection_criteria)
+            data[resource_schema[field]["new_name"]] = value
+
+    return data
+
+
+def make_resource_type_table(
+    resource_type: str,
+    schema: dict,
+    output_path: pathlib.PosixPath,
+    output_format: Literal["parquet"],
+    credential_manager: AzureFhirserverCredentialManager,
+):
+    """
+    Given a FHIR resource type, schema, and FHIR server credential manager create a
+    table containing the field from resource type specified in the the schema.
+
+    :param resource_type: A FHIR resource type.
+    :param schema: A schema specifying the desired values by FHIR resource type.
+    :param credential_manager: A credential manager for
+    a FHIR server.
+    """
+
+    output_path.mkdir(parents=True, exist_ok=True)
+    output_file_name = output_path / f"{resource_type}.{output_format}"
+
+    query = f"/{resource_type}"
+    response = query_fhir_server(credential_manager, query)
```

Comment:
This line should be moved inside of the `while` loop, and it should take the base FHIR URL as a parameter. This will allow you to remove the redundant `response = query_fhir_server(...)` call on line 149.